### PR TITLE
Update to the swagger-parser module

### PR DESCRIPTION
**Pull request by EricHenry**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by Tobias4872** [comment]
Hi Henry,

looks good to me. Much more complete now. On a side note, I have no clue how you derived the option structure. Tried to trace this down in "json-schema-ref-parser/lib/options", but did not get a handle. You obviously did, so good work.

Tobias.

**Comment by minestarks** [comment]
@EricHenry please fix the failing tests before we can merge, thank you!

**Comment by EricHenry** [comment]
@Tobias4872 the source code isn't the greatest at defining what the option structure looks like, but the docs make up for that! [check it out here](https://github.com/BigstickCarpet/swagger-parser/blob/master/docs/options.md)

@minestarks will do!

File: types/swagger-parser/index.d.ts [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
+        spec?: boolean;
+    };
+    cache?: {
+        fs?: number;
+        http?: number;
+        https?: number;
+    };
+}
+
+export function parse(api: string | any, options?: Options): Promise<any>;
```
Comment (minestarks) [inline types/swagger-parser/index.d.ts]: ```string | any``` is rather misleading, as it's the same exact type as ```any``` so it doesn't provide any stronger type checking.

Furthermore the docs state this should be a [SwaggerObject](https://github.com/BigstickCarpet/swagger-parser/blob/master/docs/swagger-parser.md#parseapi-options-callback) 

How about defining a ```Swagger``` interface and having the argument be ```string | Swagger``` ? 

If you'd rather not do that, then I'd say convert all the ```string | any```s to ```any```s.

**Review by minestarks — CHANGES_REQUESTED** [review]

File: types/swagger-parser/index.d.ts (hunk 1 above)
Comment (EricHenry) [inline types/swagger-parser/index.d.ts]: I think this is a good idea. There is an [official swagger schema](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/swagger-schema-official) type defined in the DefinitelyTyped project. Is there a supported way to reference that type, inside of this project definition?

Comment (minestarks) [inline types/swagger-parser/index.d.ts]: Yes, you can do:
```ts
import { Spec } from "swagger-schema-official";
```
See https://github.com/DefinitelyTyped/DefinitelyTyped#im-writing-a-definition-that-depends-on-another-definition-should-i-use-reference-types--or-an-import

**Review by minestarks — APPROVED** [review]

File: types/swagger-parser/index.d.ts [hunk 2]
```diff
@@ -1,7 +1,54 @@
 // Type definitions for swagger-parser 4.x
 // Project: https://www.npmjs.com/package/swagger-parser
 // Definitions by: Tobias Wolff <https://github.com/Tobias4872/>
+//                 Eric Henry <https://github.com/EricHenry/>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+import { Spec as Swagger } from '../swagger-schema-official';
```
Comment (minestarks) [inline types/swagger-parser/index.d.ts]: You'll want to change this to ```'swagger-schema-official'``` (non-relative)

Comment (sandersn) [inline types/swagger-parser/index.d.ts]: I went ahead and did this.

Comment (EricHenry) [inline types/swagger-parser/index.d.ts]: @sandersn thanks! glad I could contribute. Thanks for the guidance and helpful feedback! @minestarks

**Pull request merged**